src/popup.c

```diff
@@ -158,8 +183,11 @@ void ShowPopup(int x, int y, const char *text,
    JXSetForeground(display, rootGC, colors[COLOR_POPUP_OUTLINE]);
    JXDrawRectangle(display, popup.pmap, rootGC, 0, 0,
                    popup.width - 1, popup.height - 1);
-   RenderString(popup.pmap, FONT_POPUP, COLOR_POPUP_FG, 4, 1,
-                popup.width, popup.text);
+   for (row = 0; row < rows; row++) {
+       RenderString(popup.pmap, FONT_POPUP, COLOR_POPUP_FG, 4,
+                    (GetStringHeight(FONT_POPUP) + 1)*row+1, popup.width, multitext[row]);
```

Comment:
I think this might be a memory leak where the elements of multitext aren't released.  Just an idea, but one way around this would be to just copy the whole input string and replace the new lines with '\0's.  That would avoid some allocations too.